include/phasar/Utils/EquivalenceClassMap.h

```diff
@@ -28,7 +36,7 @@ namespace psr {
 template <typename KeyT, typename ValueT> struct EquivalenceClassMap {
   template <typename... Ts> using SetType = std::set<Ts...>;
   using EquivalenceClassBucketT = std::pair<SetType<KeyT>, ValueT>;
-  using StorageT = std::vector<EquivalenceClassBucketT>;
+  using StorageT = llvm::SmallVector<EquivalenceClassBucketT, 0>;
```

Comment:
What's the idea behind a small vector with size 0? Just to use it's interface?

Reply:
This has two reasons:
1.  `SmallVector` uses `uint32_t` for size and capacity instead of `size_t` or pointers, so it is 8 bytes smaller than `std::vector`
2. Depending on the `ValueT` type, `SmallVector` is more efficient during resizing. Since LLVM does not use exceptions by default, they do not use `noexcept`, not even for move ctors. This lets the `std::vector` copy the elements during resize instead of moving (`SmallVector` always moves). However, in this particular case, this is not really relevant, because we only put in `EdgeFunction<L>` anyway

Reply:
Hmm, that was not clear to me from the usage. Maybe, you should add a comment why this is relevant, otherwise it could be that someone changes it in the future without knowing.